File: 10/java/Vehicle.java
/*
 * The Vehicle Class.
 *
 * @author Dominic M.
 * @since 2024-10-28
 * @version 1.0
 */

/**
 * This is the Vehicle Class.
 */
public abstract class Vehicle {

    /**
     * The color of car.
     */
    private String color;

    /**
     * The speed of car.
     */
    private int speed;

    /**
     * The max speed.
     */
    private int maxSpeed;

    /**
     * The wheels of the vehicle.
     */
    private int wheels;

    /**
     * This is the constructor.
     *
     * @param colour of car
     * @param maximumSpeed of car
     * @param wheel of car
     */
    public Vehicle(
        final String colour, final int maximumSpeed, final int wheel
    ) {
        this.color = colour;
        this.speed = 0;
        this.maxSpeed = maximumSpeed;
        this.wheels = wheel;
    }

    /**
     * Getter for car color.
     *
     * @return string of car color
     */
    public String getColor() {
        return this.color;
    }

    /**
     * Sets the car color.
     *
     * @param newColor of car
     */
    public void setColor(final String newColor) {
        if (newColor != null) {
            this.color = newColor;
        } else {
            this.color = "silver";
        }
    }

    /**
     * Getter for car speed.
     *
     * @return speed of car
     */
    public int getSpeed() {
        return this.speed;
    }

    /**
     * Sets the speed.
     *
     * @param newSpeed of vehicle
     */
    public void setSpeed(final int newSpeed) {
        if (newSpeed > 0 && newSpeed < maxSpeed) {
            this.speed = newSpeed;
        }
    }

    /**
     * Getter for maxSpeed.
     *
     * @return max speed
     */
    public int getMaxSpeed() {
        return this.maxSpeed;
    }

    /**
     * Getter for wheels.
     *
     * @return amount of wheels
     */
    public int getWheels() {
      return this.wheels;
    }

    /**
     * Setter for wheels.
     *
     * @param newWheels amount
     */
    public void setWheels(final int newWheels) {
      if (newWheels > 0) {
        this.wheels = newWheels;
      }
    }

    /**
     * This method accelerates the car.
     *
     * @param accelPower how much force is being put into acceleration
     * @param accelTime how long it is accelerating for
     */
    public void accelerate(final int accelPower, final int accelTime) {
        speed = (accelPower * accelTime) + speed;
        if (this.speed > maxSpeed) {
            this.speed = maxSpeed;
        }
    }

    /**
     * This method brakes the car.
     *
     * @param brakePower how much force is being put into the brakes
     * @param brakeTime how long it is braking for
     */
    public void brake(final int brakePower, final int brakeTime) {
        this.speed = this.speed - (brakePower * brakeTime);
        if (this.speed < 0) {
          this.speed = 0;
        }
    }
}

File: 10/ts/Vehicle.js
"use strict";
/**
 * The Vehicle Class
 *
 * author Dominic M.
 * version 1.0
 * since 2024-10-28
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vehicle = void 0;
var Vehicle = /** @class */ (function () {
    function Vehicle(color, maxSpeed, wheels) {
        this.color = color;
        this.speed = 0;
        this.maxSpeed = maxSpeed;
        this.wheels = wheels;
    }
    Object.defineProperty(Vehicle.prototype, "theColor", {
        /**
         * Getter for car color
         *
         * @returns {string} of car color
         */
        get: function () {
            return this.color;
        },
        /**
         * Sets the car color
         *
         * @param {string} newColor of car
         */
        set: function (newColor) {
            if (newColor != null) {
                this.color = newColor;
            }
            else {
                this.color = 'silver';
            }
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(Vehicle.prototype, "theSpeed", {
        /**
         * Getter for car speed
         *
         * @returns {number} of car speed
         */
        get: function () {
            return this.speed;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(Vehicle.prototype, "theMaxSpeed", {
        /**
         * Getter for max speed
         *
         * @returns {number} of max car speed
         */
        get: function () {
            return this.maxSpeed;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(Vehicle.prototype, "theWheels", {
        /**
         * Getter for wheels
         *
         * @return {number} of wheels
         */
        get: function () {
            return this.wheels;
        },
        enumerable: false,
        configurable: true
    });
    /**
     * This method accelerates the car
     *
     * @param {number} accelPower
     * @param {number} accelTime
     */
    Vehicle.prototype.accelerate = function (accelPower, accelTime) {
        this.speed = (accelPower * accelTime) + this.speed;
        if (this.speed > this.maxSpeed) {
            this.speed = this.maxSpeed;
        }
    };
    /**
     * This method decelerates the car
     *
     * @param {number} breakPower
     * @param {number} breakTime
     */
    Vehicle.prototype.break = function (breakPower, breakTime) {
        this.speed = this.speed - (breakPower * breakTime);
        if (this.speed < 0) {
            this.speed = 0;
        }
    };
    return Vehicle;
}());
exports.Vehicle = Vehicle;
